### PR TITLE
Query proof with specific height parameter

### PR DESCRIPTION
**Pull request by prin-r**
## Implementation details

You can now ask for proof by specifying the block height by entering it as a parameter of url.
For example: 
    http://localhost:1317/oracle/proof/1?height=997

Note that with default pruning schemes, you can ask historical proof only 100 blocks from the latest block.
For example (latest block = 1555): 
    http://localhost:1317/oracle/proof/1?height=1496 (1555 - 1496 < 100)
    http://localhost:1317/oracle/proof/1?height=1542 (1555 - 1542 < 100)
    etc

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: chain/x/oracle/client/common/proof/proof.go [hunk 1]
```diff
@@ -92,8 +92,17 @@ type Proof struct {
 	EVMProofBytes tmbytes.HexBytes `json:"evmProofBytes"`
 }
 
-func GetProofHandlerFn(cliCtx context.CLIContext, route string) http.HandlerFunc {
+func GetProofHandlerFn(ctx context.CLIContext, route string) http.HandlerFunc {
```
Comment (taobun) [inline chain/x/oracle/client/common/proof/proof.go]: We didn't rename this param

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/client/common/proof/proof.go (hunk 1 above)
Comment (prin-r) [reply chain/x/oracle/client/common/proof/proof.go]: done

File: chain/x/oracle/client/common/proof/proof.go [hunk 2]
```diff
@@ -243,3 +263,178 @@ func GetProofHandlerFn(cliCtx context.CLIContext, route string) http.HandlerFunc
...
+			return
+		}
+		height := &ctx.Height
+		if ctx.Height == 0 {
+			height = nil
+		}
+
+		commitHeight := uint64(0)
+		requestIDs := r.URL.Query()["id"]
+		fmt.Println(requestIDs)
```
Comment (taobun) [inline chain/x/oracle/client/common/proof/proof.go]: Remove this

File: chain/x/oracle/client/common/proof/proof.go [hunk 3]
```diff
@@ -243,3 +263,178 @@ func GetProofHandlerFn(cliCtx context.CLIContext, route string) http.HandlerFunc
...
+			if err := ctx.Codec.UnmarshalJSON(qResult.Result, &request); err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			if request.Result == nil {
+				rest.WriteErrorResponse(w, http.StatusNotFound, "Result has not been resolved")
+				return
+			}
+
+			commit, err := ctx.Client.Commit(height)
```
Comment (taobun) [inline chain/x/oracle/client/common/proof/proof.go]: Bring this to out of loop.

File: chain/x/oracle/client/common/proof/proof.go [hunk 4]
```diff
@@ -243,3 +263,178 @@ func GetProofHandlerFn(cliCtx context.CLIContext, route string) http.HandlerFunc
...
+				opType := op.GetType()
+				if opType == "iavl:v" {
+					err := ctx.Codec.UnmarshalBinaryLengthPrefixed(op.GetData(), &iavlProof)
+					if err != nil {
+						rest.WriteErrorResponse(w, http.StatusInternalServerError,
+							fmt.Sprintf("iavl: %s", err.Error()),
+						)
+						return
+					}
+				} else if opType == "multistore" {
```
Comment (taobun) [inline chain/x/oracle/client/common/proof/proof.go]: Only save this value to process later

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/client/common/proof/proof.go (hunk 2 above)
Comment (prin-r) [reply chain/x/oracle/client/common/proof/proof.go]: done

**Review by taobun — APPROVED** [review]
LGTM

**Pull request merged**